### DISCUSSION TITLE
Simulating the impact of modified capacities of energy sources

### DISCUSSION THREAD

**Discussion by brmarconi** [electricitymaps/electricitymaps-contrib#4876] [Ideas]
I am pretty sure it has already been proposed or brain-stormed... Would it be feasible to expose a GUI (and/or API)  to simulate the impact on the carbon footprint of modified capacities of energy sources. The simulator would use input such as increased installed capacity of some sources (typically carbon free) and decreased capacity of some other (typically high carbon), and cross this input with historical data (from which would be deduced the knowledge of source capacity factors (**)), to evaluate the impact on carbon emission.

The simulator could also evaluate which energy source replacement to prioritize, or compute how much the capacity of source X would have to be increased if source Y's is reduced, etc.

(**) I am assuming that given the capacity of a source (e.g. solar) and of its production across the years, it becomes possible to estimate its approximate capacity factor for a given zone.

**Comment by VIKTORVAV99** [comment]
I could very well work for a country or zone when in production mode but since the consumption mode is calculated based on production + import, and the import of those imports and so on this would pretty much be impossible to do with any accuracy in the consumption mode (using the app).

So while it's a great idea I don't think it's something we can implement, at least not right now, but if you have any ideas on how this could be simplified please do share those ideas.

**Reply by brmarconi** [reply]
Can't an import be considered as any other source, albeit with a highly variable carbon intensity? I'd assume a model where sources include production plus imports, and where the characteristics of a source are: capacity, production over a period of time, carbon intensity over a period of time (plus label == type or country). Then the model is exercised by doubling, say, the capacity of a source (type or country), and using historical data to get the capacity factor for that source and in the case of a country its average carbon intensity across a selected period of time. Such as, what is the impact on France and Germany's carbon intensities when the capacity of the exchange across those countries is doubled? (for a given period of time, assuming repeatability of past exchanges across France and Germany). 
Too naive?

**Reply by VIKTORVAV99** [reply]
Sadly I do think it's a little too naïve as there are several factors that would make it hard to get a good estimate of such a model.

First we have the impact of the added capacity itself. If we want to model how it would impact a zone in the future we also need to model the growth or decline of consumption, growth or decline of other production sources. And this is before we involve imports and market consitions.

Lets use solar as an example where we can predict the electricity generation over a year with pretty high accuracy.

It would be pretty easy to just assign more production to solar using the average capacity factor of the previous years but this would risk being highly miss representative for a few reasons.

It might be that it's simply not possible to add such high amounts of solar to the current grid, both for market conditions and infrastructure conditions. It might be that a increase in solar leads to high exports during the day and high imports during the night making the addition affect other zones as well causing cascading impacts.

There is also the possibility of adding solar would lower the generation of other sources and that would be hard to estimate ahead of time.

Some of these issues would be removed if we only allowed this during production mode, such as the import/export and the effect on other zones.

A more technical issue about all of this is we don't yet have all the data we would need, like historical capacity data and the teams are currently busy working on other features.

It's definitely an interesting idé though but I'm not necessarily sure it's something for Electricity Maps at this time.

**Reply by brmarconi** [reply]
> First we have the impact of the added capacity itself. If we want to model how it would impact a zone in the future we also need to model the growth or decline of consumption, growth or decline of other production sources. 

Right, I oversimplified my explanation. One major goal of a model would be to evaluate if/how/where high carbon intensity sources can be displaced. 

> Lets use solar as an example where we can predict the electricity generation over a year with pretty high accuracy.
> 
> It would be pretty easy to just assign more production to solar using the average capacity factor of the previous years but this would risk being highly miss representative for a few reasons.
> 
> It might be that it's simply not possible to add such high amounts of solar to the current grid, both for market conditions and infrastructure conditions.

A model, at least in a first iteration, would exclude those conditions, because its goal is to provide a broad direction.

> It might be that a increase in solar leads to high exports during the day and high imports during the night making the addition affect other zones as well causing cascading impacts.

Yes but I was guessing either walking the chain of imports to get at the source, of average somehow the imports. 

> There is also the possibility of adding solar would lower the generation of other sources and that would be hard to estimate ahead of time.

These could be inputs to the model, at least initially.

> Some of these issues would be removed if we only allowed this during production mode, such as the import/export and the effect on other zones.
> 
> A more technical issue about all of this is we don't yet have all the data we would need, like historical capacity data and the teams are currently busy working on other features.
> 
> It's definitely an interesting idé though but I'm not necessarily sure it's something for Electricity Maps at this time.

OK, thank you very much for the discussion.
Related question though, it such a model something that could be implemented separately using data produced by your APIs?

**Reply by VIKTORVAV99** [reply]
With the commercial API it should be possible if you can find another source for the historical capacity to calculate the capacity factors for each source. I'll add a note and check in with the team after the holidays if we can get these values added to the API as well but it would probably take a while as we would have to modify our existing parsers or build new ones to get the capacity.

You can check out https://static.electricitymaps.com/api/docs/index.html#routes to see what the different endpoints return.
